tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/action/WatchTaskAction.java

```diff
@@ -133,11 +147,21 @@ public void done(IJobChangeEvent event) {
 								}
 							}
 						}
-						catch (IOException e) {
-							GradleUI.logError("Could not uninstall bundles installed by watch task", e);
+						catch (IOException ioe) {
+							GradleUI.logError("Could not uninstall bundles installed by watch task", ioe);
 						}
 					}
 
+					@Override
+					@SuppressWarnings("deprecation")
+					public void running(IJobChangeEvent event) {
+						WatchDecorator lightweightLabelDecorator =
+							(WatchDecorator)decoratorManager.getLightweightLabelDecorator(
```

Comment:
hey greg, I will investigate it

Reply:
hi greg, I use decoratorManager.update to replace manually refresh.

Reply:
Right but it shouldn't be necessary, there should be a more portable way to do it.  Let me know if you need to help in researching it.

Reply:
because watch don't depends on the server we create inside the eclipse. So I feel we can't let server view to trigger UI refresh to update decorator. I have posted a new pull to terry by simply calling decoratorManager.update to refresh decorator.  He will be forward to you.